src/ophys_etl/modules/segmentation/graph_utils/feature_vector_rois.py

```diff
@@ -0,0 +1,522 @@
+        upper left corner of whatever frame is represented by sub_video)
+
+    filter_fraction: float
+        The fraction of brightest timesteps to be used in calculating
+        the Pearson correlation between pixels
+
+    rng: Optional[np.random.RandomState]
+        A random number generator used to choose pixels which will be
+        used to select the brightest filter_fraction of pixels. If None,
+        an np.random.RandomState will be instantiated with a hard-coded
+        seed (default: None)
+
+    pixel_ignore: Optional[np.ndarray]:
+        An (n_rows, n_cols) array of booleans marked True at any pixels
+        that should be ignored, presumably because they have already been
+        selected as ROI pixels. (default: None)
+
+    Returns
+    -------
+    features: np.ndarray
+         An (n_rows*n_cols, n_rows*n_cols) array of feature vectors.
+         features[ii, :] is the feature vector for the ii_th pixel
+         in sub_video (where ii can be mapped to (row, col) coordinates
+         using np.unravel_index()
+
+    Notes
+    -----
+    The feature vectors returned by this method are calculated as follows:
+
+    1) Select a random set of 10 pixels from sub_video including the seed_pt.
+
+    2) Use the brightest filter_fraction of timestamps from the 10 selected
+    seeds to calculate the Pearson correlation coefficient between each
+    pair of pixels. This results in an (n_pixels, n_pixels) array in which
+    pearson[ii, jj] is the correlation coefficient between the ii_th and
+    jj_th pixels.
+
+    3) For each column pearson [:, ii] in the array of Pearson
+    correlation coefficients, subtract off the minimum value in
+    that column and divide by the gap between the 25th adn 75th
+    percentile of that column's values.
+
+    Each row in the matrix of Pearson correlation coefficients is now
+    a feature vector corresponding to that pixel in sub_video.
+    """
+
+    # fraction of timesteps to discard
+    discard = 1.0-filter_fraction
+
+    n_rows = sub_video.shape[1]
+    n_cols = sub_video.shape[2]
+
+    # start assembling mask in timesteps
+    trace = sub_video[:, seed_pt[0], seed_pt[1]]
+    thresh = np.quantile(trace, discard)
+    global_mask = []
+    mask = np.where(trace >= thresh)[0]
+    global_mask.append(mask)
+
+    # choose n_seeds other points to populate global_mask
+    i_seed = np.ravel_multi_index(seed_pt, sub_video.shape[1:])
+    possible_seeds = []
+    for ii in range(n_rows*n_cols):
+        if ii == i_seed:
+            continue
+        p = np.unravel_index(ii, sub_video.shape[1:])
+
+        if pixel_ignore is None or not pixel_ignore[p[0], p[1]]:
+            possible_seeds.append(ii)
+
+    n_seeds = 10
+    if rng is None:
+        rng = np.random.RandomState(87123)
+    chosen = set()
+    chosen.add(seed_pt)
+
+    if len(possible_seeds) > n_seeds:
+        chosen_dex = rng.choice(possible_seeds,
```

Comment:
it is not clear to me what `dex` stand for

Reply:
Personal abbreviation of "index". I can see how it is not obvious. I'll spell it out.